x-pack/metricbeat/module/statsd/server/registry.go

```diff
@@ -69,6 +68,118 @@ func (d *deltaGaugeMetric) Value() float64 {
 	return d.value
 }
 
+// SamplingTimer is a timer that supports sampling
+type samplingTimer struct {
+	metrics.Timer
+	meter     metrics.Meter
+	histogram metrics.Histogram
+}
+
+// NewSamplingTimer returns a new SamplingTimer
+func newSamplingTimer() *samplingTimer {
+	m := metrics.NewMeter()
+	h := metrics.NewHistogram(metrics.NewExpDecaySample(1028, 0.015))
+
+	return &samplingTimer{
+		Timer:     metrics.NewCustomTimer(h, m),
+		meter:     m,
+		histogram: h,
+	}
+}
+
+// SampledUpdate will update the timer a sampled measurement
+func (s *samplingTimer) SampledUpdate(d time.Duration, sampleRate float64) {
```

Comment:
I see that the `StandardTimer` uses a lock for `Update`, should we also use one here? I guess that we are not updating these metrics in multiple threads now, but just in case in the future we do it and we forget about this.

If we do that we may also need to implement `Update` and `UpdateSince` to avoid using different locks.

Reply:
True, however there's lots more that will break if we suddenly use this in a multithreaded scenario. samplingTimer is not exported, so no one else can use it. I'd rather like to avoid sprinkling locks in this code for now.

Reply:
:+1: